compiler-js/src/test/ceylon/dyns/issues_dyns.ceylon

```diff
@@ -330,3 +331,9 @@ void issue5959() {
     check((d of Object) is C5959<String>, "#5959 targs3");
   }
 }
+
+void issue6057() {
+  dynamic DefaultObject {}
+  dynamic { DefaultObject o = eval("this"); }
```

Comment:
@chochos the purpose is to get the "default" object (I don't know the correct name for this sorry) which is `window` in the browser and `process` in nodejs. The infinite recursion only occurs if the default object is `dre$$`ed with the default implementation of the hash attribute. I have verified that my code reproduces the problem when running in nodejs and the browser (without the fix in place). The nodejs exception was something like "RangeError: ... recursion ...".